render/texture.h

```diff
@@ -1,43 +1,48 @@
 #pragma once
 
+#include <GL/glew.h>
+
 #include <string>
+#include <vector>
 
 namespace pong
 {
 
 class Texture
 {
 public:
+
+    static Texture CreateFromFile(const std::string& filePath);
+    /* Assumes the data is grayscale and stored from top left to bottom right in vector */
+    static Texture CreateFromFontCharacter(const std::vector<unsigned char>& imageData, int width, int height);
+    static Texture CreateFromSolidColor(unsigned char r, unsigned char g, unsigned char b, unsigned char a);
```

Comment:
Curious if you have any thoughts (positives, negatives) on this kind of API. I didn't want to keep inheriting from this class like I did for SolidColorTexture. ie some kind of `FontTexture` for create from font bitmaps specfically. So instead I thought these static creation functions could avoid that. 

The only thing that is a little weird is how you cant really create smart pointers of a Texture easily, unless I were to create new static functions that would specifcally return that?

Reply:
Hmm, great question.

I think there are many ways to go about this process, so know this isn't the only way you could do it.

I would start by making an rgba struct, seeing as that's a pretty common thing here, and you can simplify the rest of your code with it.

From there, I would do the following:

* Drop the static methods, just use constructors, aka `Texture(const std::vector<unsigned char>& imageData, int width, int height)` and `Texture(const Rgba& color);`

from there, you can either call the constructor directly, or now make_unique or make_shared will defer to these constructors depending on their arguments.


I think despite not having a "nicely named" static function, the ability to naturally use make_shared or make_unique is a good simplification which doesn't have you exploding the complexity of:

static Texture CreateFromFontCharacter(const std::vector<unsigned char>& imageData, int width, int height);

// put this in a shared or unique pointer!!
static Texture* AllocateFromFontCharacter(const std::vector<unsigned char>& imageData, int width, int height);

and so on.


I think using the existing facilities may be an improvement

Reply:
Yeah that is a good point.